# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CompositorBridgeParent.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 gfx/layers/ipc/CompositorBridgeParent.h
--- a/gfx/layers/ipc/CompositorBridgeParent.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/gfx/layers/ipc/CompositorBridgeParent.h	Wed Jan 08 22:19:14 2020 +0000
@@ -15,10 +15,11 @@
 //       which the deadline will be 15ms + throttle threshold
 //#define COMPOSITOR_PERFORMANCE_WARNING
 
-#include <stdint.h>              // for uint64_t
-#include "Layers.h"              // for Layer
-#include "mozilla/Assertions.h"  // for MOZ_ASSERT_HELPER2
-#include "mozilla/Attributes.h"  // for override
+#include <stdint.h>                   // for uint64_t
+#include "Layers.h"                   // for Layer
+#include "mozilla/Assertions.h"       // for MOZ_ASSERT_HELPER2
+#include "mozilla/Attributes.h"       // for override
+#include "mozilla/GfxMessageUtils.h"  // for WebGLVersion
 #include "mozilla/Maybe.h"
 #include "mozilla/Monitor.h"    // for Monitor
 #include "mozilla/RefPtr.h"     // for RefPtr
@@ -53,10 +54,11 @@
 
 class CancelableRunnable;
 
-namespace webgpu {
-class PWebGPUParent;
-class WebGPUParent;
-}  // namespace webgpu
+namespace dom {
+class HostWebGLCommandSink;
+class HostWebGLErrorSource;
+class WebGLParent;
+}  // namespace dom
 
 namespace gfx {
 class DrawTarget;
@@ -71,6 +73,11 @@
 #endif
 }  // namespace ipc
 
+namespace webgpu {
+class PWebGPUParent;
+class WebGPUParent;
+}  // namespace webgpu
+
 namespace layers {
 
 class APZCTreeManager;
@@ -297,6 +304,16 @@
     return IPC_FAIL_NO_REASON(this);
   }
 
+  virtual PWebGLParent* AllocPWebGLParent(
+      const WebGLVersion& aVersion, const HostWebGLCommandSink& aCommandSink,
+      const HostWebGLErrorSource& aErrorSource) = 0;
+
+  virtual PWebGLParent* AllocPWebGLParent(
+      const WebGLVersion& aVersion, HostWebGLCommandSink&& aCommandSink,
+      HostWebGLErrorSource&& aErrorSource) = 0;
+
+  virtual bool DeallocPWebGLParent(PWebGLParent* aWebGLParent) = 0;
+
   bool mCanSend;
 
  private:
@@ -683,8 +700,29 @@
 #endif  // defined(MOZ_WIDGET_ANDROID)
 
   WebRenderBridgeParent* GetWrBridge() { return mWrBridge; }
+  webgpu::WebGPUParent* GetWebGPUBridge() { return mWebGPUBridge; }
 
-  webgpu::WebGPUParent* GetWebGPUBridge() { return mWebGPUBridge; }
+  PWebGLParent* AllocPWebGLParent(
+      const WebGLVersion& aVersion, const HostWebGLCommandSink& aCommandSink,
+      const HostWebGLErrorSource& aErrorSource) override {
+    MOZ_ASSERT_UNREACHABLE(
+        "This message is CrossProcessCompositorBridgeParent only");
+    return nullptr;
+  }
+
+  PWebGLParent* AllocPWebGLParent(
+      const WebGLVersion& aVersion, HostWebGLCommandSink&& aCommandSink,
+      HostWebGLErrorSource&& aErrorSource) override {
+    MOZ_ASSERT_UNREACHABLE(
+        "This message is CrossProcessCompositorBridgeParent only");
+    return nullptr;
+  }
+
+  bool DeallocPWebGLParent(PWebGLParent* aWebGLParent) override {
+    MOZ_ASSERT_UNREACHABLE(
+        "This message is CrossProcessCompositorBridgeParent only");
+    return false;
+  }
 
  private:
   void Initialize();